docs/src/explanations/interfaces-and-connections.md

```diff
@@ -81,10 +81,51 @@ class Block extends Module {
 ```
 where <> bulk connects interfaces of opposite gender between sibling modules or interfaces of the same gender between parent/child modules.
 
-Bulk connections connect leaf ports of the same name to each other. If the names do not match or are missing, Chisel does not generate a connection.
+The bidirectional bulk connection operator `<>` connects leaf ports of the same name to each other. The Scala types of the Bundles are not required to match. If one named signal is missing from either side, Chisel will give an error such as in the following example:
 
+```scala mdoc:silent
+
+class NotReallyAFilterIO extends Bundle {
+  val x = Flipped(new PLink)
+  val y = new PLink
+  val z = Output(new Bool())
+}
+class Block2 extends Module {
+  val io1 = IO(new FilterIO)
+  val io2 = IO(Flipped(new NotReallyAFilterIO))
+
+  io1 <> io2
+}
+```
+Below we can see the resulting error for this example:
+```scala mdoc:crash
+ChiselStage.emitVerilog(new Block2)
+```
 Caution: bulk connections should only be used with **directioned elements** (like IOs), and is not magical (e.g. connecting two wires isn't supported since Chisel can't necessarily figure out the directions automatically [chisel3#603](https://github.com/freechipsproject/chisel3/issues/603)).
 
+For example, putting two temporary wires and connecting them here will not work, even though the directions could be known from the endpoints:
+
+```scala mdoc:silent
+
+class BlockWithTemporaryWires extends Module {
+  val io = IO(new FilterIO)
+  val f1 = Module(new Filter)
+  val f2 = Module(new Filter)
+  f1.io.x <> io.x
+ val tmp1 = Wire(new FilterIO)
+ val tmp2 = Wire(new FilterIO)
+  f1.io.y <> tmp1
+  tmp1 <> tmp2
+  tmp2 <> f2.io.x
+  f2.io.y <> io.y
+}
+
+```
+Below we can see the resulting error for this example:
+```scala mdoc:crash
+ChiselStage.emitVerilog(new BlockWithTemporaryWires)
+```
+For more details and information, see [Deep Dive into Connection Operators](Connection-operators-.md)
 ## The standard ready-valid interface (ReadyValidIO / Decoupled)
```

Comment:
```suggestion

NOTE: When using `Chisel._` (compatibility mode) instead of `chisel3._`, the `:=` operator works in a bidirectional fashion similar to `<>`, but not exactly the same.

## The standard ready-valid interface (ReadyValidIO / Decoupled)
```